app/controllers/fhir/r4/api_controller.rb

```diff
@@ -457,6 +459,27 @@ def current_client_application
     OauthApplication.find_by(id: doorkeeper_token.application_id) if doorkeeper_token.application_id.present?
   end
 
+  # Determine if the patient's jurisdiction is valid for the requesting application
+  def jurisdiction_valid_for_client?(patient)
+    allowed_jurisdiction_ids = current_client_application.jurisdiction&.subtree&.pluck(:id) ||
+                               current_resource_owner&.jurisdiction&.subtree&.pluck(:id)
+
+    if allowed_jurisdiction_ids.nil?
+      patient.errors.add(:jurisdiction_id, 'Client application does not have a jurisdiction')
+      false
+    elsif !patient.jurisdiction.nil? && allowed_jurisdiction_ids.include?(patient.jurisdiction[:id])
+      true
+    else
+      patient.errors.add(:jurisdiction_id, "Jurisdiction must be within the client application's jurisdiction")
+      false
+    end
+  end
+
+  # If no jurisdiction specified for a patient, default to the client's jurisdiction
+  def default_patient_jurisdiction
+    current_client_application.jurisdiction || current_resource_owner&.jurisdiction
```

Comment:
Same comment as above related to priority between user workflow and m2m workflow.